### PR TITLE
Remove NOAA solar cycle sample data

### LINKED ISSUE #4156: Remove NOAA solar cycle data from sample data set
### Description

NOAA solar cycle data will always be out of date given that it is always updating. This is also true of the prediction data set. The names of these data sets are `NOAAINDICES_TIMESERIES` and `NOAAPREDICT_TIMESERIES`. Given that these are simply available as csv files, ftp://ftp.swpc.noaa.gov/pub/weekly/Predict.txt and ftp://ftp.swpc.noaa.gov/pub/weekly/RecentIndices.txt, that are easily using the existing clients [NOAAIndicesClient](https://docs.sunpy.org/en/stable/api/sunpy.net.dataretriever.NOAAIndicesClient.html#sunpy.net.dataretriever.NOAAIndicesClient) and NOAAPredictIndicesClient, I suggest that these be removed from the sample data set.

### PR DESCRIPTION
**Pull request by ehsteve**
### Description

Fixes #4156 

This PR needs to be coordinated with PR #4135. I did not touch that particular example so as not to create a conflict.

### CODE REVIEW

File: examples/time_series/timeseries_convolution_filter.py [hunk 1]
```diff
@@ -11,29 +11,28 @@
...
-    'sunspot SWO Smoothed',
-    convolve(ts_noaa_ind.quantity('sunspot SWO'), kernel=Box1DKernel(10))
+goes_lc = goes_lc.add_column(
+    'xrsa_smoothed',
+    convolve(goes_lc.quantity('xrsa'), kernel=Box1DKernel(50))
 )
 
 ###############################################################################
 # Plotting original and smoothed timeseries
-plt.ylabel('Sunspot Number')
```
Comment (hayesla) [inline examples/time_series/timeseries_convolution_filter.py:33]: `plt.ylabel(Flux (Wm$^{-2}$)`

Comment (hayesla) [inline examples/time_series/timeseries_convolution_filter.py:33]: `plt.ylabel('Flux Wm$^{-2}$)')`

File: examples/time_series/timeseries_peak_finding.py [hunk 2]
```diff
@@ -97,11 +101,10 @@ def findpeaks(series, DELTA):
 # Now we take the column 'sunspot SWO' of this TimeSeries and try to find it's
 # extrema using the function findpeaks. We take the value of DELTA to be
 # approximately the length of smallest peak that we wish to detect.
-series = my_timeseries.to_dataframe()['sunspot SWO']
-minpeaks, maxpeaks = findpeaks(series, DELTA=10.)
+series = my_timeseries.to_dataframe()['xrsa']
+minpeaks, maxpeaks = findpeaks(series, DELTA=1e-7)
 # Plotting the figure and extremum points
 fig, ax = plt.subplots()
-ax.set_ylabel('Sunspot Number')
```
Comment (hayesla) [inline examples/time_series/timeseries_peak_finding.py:104]: `ax.set_ylabel('Flux (Wm$^{-2}$)')`

**Comment by ehsteve** [comment]
@Cadair not that it really matters but the file can be removed from the sample data server and a release of the sample data repo should be made in coordination with the next release which would include this PR.

**Comment by nabobalis** [comment]
Add a changelog.

**Comment by nabobalis** [comment]
Fix the comments.

File: sunpy/timeseries/sources/noaa.py [hunk 3]
```diff
@@ -36,8 +36,7 @@ class NOAAIndicesTimeSeries(GenericTimeSeries):
     Examples
     --------
     >>> import sunpy.timeseries
-    >>> import sunpy.data.sample  # doctest: +REMOTE_DATA
-    >>> noaa = sunpy.timeseries.TimeSeries(sunpy.data.sample.NOAAINDICES_TIMESERIES, source='NOAAIndices')  # doctest: +REMOTE_DATA
+    >>> noaa = sunpy.timeseries.TimeSeries("ftp://ftp.swpc.noaa.gov/pub/weekly/RecentIndices.txt", source='NOAAIndices')  # doctest: +REMOTE_DATA
```
Comment (ehsteve) [inline sunpy/timeseries/sources/noaa.py:39]: @hayesla Is this weird? I didn't think it made sense to include a call to Fido in the example section of a doc string.

Comment (hayesla) [reply sunpy/timeseries/sources/noaa.py:39]: I actually think it's a nice example of how TimeSeries can also accept a URL!:)

Comment (hayesla) [reply sunpy/timeseries/sources/noaa.py:39]: I actually think its a nice example of how TimeSeries can also accept a URL! :)

**Comment by wtbarnes** [comment]
Does this need to be milestoned for 2.0?

**Comment by nabobalis** [comment]
Why not? #YOLO

**Comment by ehsteve** [comment]
I merged PR #4135  into this one.

**Comment by ehsteve** [comment]
> Add a changelog.

Done.

File: examples/plotting/solar_cycle_example.py [hunk 4]
```diff
@@ -3,24 +3,38 @@
...
+# sunspot record from 2015 and prediction for the cycle till 2019 from (SWPC)
+# part of the National Oceanic and Atmospheric Administration (NOAA).
+# This code snippet downloads and grabs the NOAA solar cycle data as a
+# ``TimeSeries``.
+result = Fido.search(a.Time("2015/1/1", datetime.datetime.now()),
+                     a.Instrument('noaa-indices'))
+f_noaa_indices = Fido.fetch(result)
+result = Fido.search(a.Time(datetime.datetime.now(), datetime.datetime.now() +
+                     datetime.timedelta(days=4*365)),
+                     a.Instrument(''noaa-predict''))
```
Comment (dstansby) [inline examples/plotting/solar_cycle_example.py:27]:
```suggestion
                     a.Instrument('noaa-predict'))
```

**Comment by ehsteve** [comment]
> Looks like there's still somewhere in the doc build trying to use the deleted sample data: https://app.circleci.com/pipelines/github/sunpy/sunpy/5697/workflows/da48cbc5-c085-46e2-8354-94fbd712cee3/jobs/51223/steps

That's the solar_cycle_example which should be fixed now.

**Comment by mbobra** [comment]
Maybe I am missing something, but I'm a little confused about the explanatory text.

Here is the first paragraph:

> For this example we will downaload sunpy's data which has an old sunspot record from 2015 and prediction for the cycle till 2019 from (SWPC) part of the National Oceanic and Atmospheric Administration (NOAA). This code snippet downloads and grabs the NOAA solar cycle data as a ``TimeSeries``.

I'm having a little trouble understanding the text. First, we're not downloading SunPy's data, we're downloading NOAA data using Fido, right? And can we clarify what two things we're downloading, because I'm still confused about the distinction between `noaa` and `noaa_predict`, as explained in the code (below):

> Finally, we plot both ``noaa`` and ``noaa_predict`` together, with an arbitrary range for the strength of the next solar cycle. Here ``noaa`` refers to existing solar cycle data from 2015 and ``noaa_predict`` is a prediction for the rest of the solar cycle till 2019.

Perhaps it might be useful to give a little more introductory material. What is the predicted sunspot number [(perhaps link to the documentation)](https://docs.sunpy.org/en/stable/api/sunpy.timeseries.sources.NOAAPredictIndicesTimeSeries.html)? How does NOAA come up with it? Why is it useful? One sentence can address all of these questions. It might be nice to fix up the spelling and grammar too (we should spell out all words like "until" instead of "till"). [Here is a link](https://docs.astropy.org/en/stable/development/style-guide.html) to the AstroPy style guide.

**Comment by ehsteve** [comment]
@Cadair @nabobalis I don't understand these failures.

**Comment by nabobalis** [comment]
> @Cadair @nabobalis I don't understand these failures.

The doc build fail says

```
result = Fido.search(a.Time("2015/1/1", datetime.datetime.now()),
 
NameError: name 'datetime' is not defined
```

You also need to rebase the PR or merge in master.

**Comment by ehsteve** [comment]
@mbobra @nabobalis sorry I thought that the last PR was finished and passing so didn't look at that file. I've made some update there including to the text to make clearer based on @mbobra comments. Unfortunately I can't find much information about these values and frankly I think that references like you are asking for here deserve to be added to the doc string for these time series and not live in the example. In any case, let me know what you think of my changes. I guess I've taken over on this.

File: examples/plotting/solar_cycle_example.py [hunk 5]
```diff
@@ -1,33 +1,53 @@
...
-# Finally, we plot both ``noaa`` and ``noaa2`` together, with an arbitrary
-# range for the strength of the next solar cycle.
+# Finally, we plot both ``noaa`` and ``noaa_predict`` for the sunspot number.
+# In this case we use the S.I.D.C. Brussels International Sunspot Number (RI).
+# The predictions provide both a high and low values which we plot below as
+# ranges.
 fig, ax = plt.subplots()
-ax.plot(noaa.data.index, noaa.data['sunspot RI'], label='Sunspot Number')
-ax.plot(noaa_predict.data.index, noaa_predict.data['sunspot'],
+ax.plot(noaa.index, noaa.quantitu('sunspot RI'), label='Sunspot Number')
```
Comment (hayesla) [inline examples/plotting/solar_cycle_example.py:44]:
```suggestion
ax.plot(noaa.index, noaa.quantity('sunspot RI'), label='Sunspot Number')
```

File: examples/plotting/solar_cycle_example.py [hunk 6]
```diff
@@ -1,33 +1,53 @@
...
+###############################################################################
+# The U.S. Dept. of Commerce, NOAA, Space Weather Prediction Center (SWPC)
+# provides recent solar cycle indices which includes different sunspot numbers,
+# radio flux, and geomagnetic index. They also provide predictions for how the
+# sunspot number and radio flux will evolve. Predicted values are based on the
+# consensus of the Solar Cycle 24 Prediction Panel.
+#
+# We will first search for and then download the data.
+time_range = TimeRange("2008-06-01 00:00", Time.now())
+result = Fido.search(a.Time(time_range.start, time_range.end), a.Instrument('noaa-indices'))
```
Comment (hayesla) [inline examples/plotting/solar_cycle_example.py:27]: `
```suggestion
result = Fido.search(a.Time(time_range), a.Instrument('noaa-indices'))
```

Comment (hayesla) [reply examples/plotting/solar_cycle_example.py:27]: or was this .start and .end added for a reason?

Comment (nabobalis) [reply examples/plotting/solar_cycle_example.py:27]: Do we need a time range for the search?

File: examples/time_series/timeseries_convolution_filter.py [hunk 7]
```diff
@@ -9,32 +9,31 @@
...
 plt.xlabel('Time')
+plt.ylabel("Flux (Wm$^{-2}$")
 plt.title('Smoothing of Time Series')
-plt.plot(ts_noaa_ind.quantity('sunspot SWO'), label='original data')
-plt.plot(ts_noaa_ind.quantity('sunspot SWO Smoothed'), label='smoothed')
+goes_lc.to_dataframe()['xrsa'].plot(label='original')
```
Comment (hayesla) [inline examples/time_series/timeseries_convolution_filter.py:36]:
```suggestion
plt.plot(goes_lc.quantity('xrsa', label='original')
```

File: examples/time_series/timeseries_convolution_filter.py [hunk 8]
```diff
@@ -9,32 +9,31 @@
...
+goes_lc.to_dataframe()['xrsa_smoothed'].plot(label='smoothed')
```
Comment (hayesla) [inline examples/time_series/timeseries_convolution_filter.py:37]:
```suggestion
plt.plot(goes_lc.quantity('xrsa_smoothed', label='smoothed')
```

File: examples/plotting/solar_cycle_example.py (hunk 6 above)
Comment (hayesla) [reply examples/plotting/solar_cycle_example.py:27]: Yes I think you need to include a time for a Fido search no?

**Comment by mbobra** [comment]
Thanks @ehsteve, it looks much better now!

File: examples/plotting/solar_cycle_example.py (hunk 6 above)
Comment (nabobalis) [reply examples/plotting/solar_cycle_example.py:27]: I thought that was removed. Maybe I'm wrong.

**Comment by ehsteve** [comment]
Thanks @hayesla!

**Comment by mbobra** [comment]
I don't understand this doc build error:

```python
  File "/root/project/examples/plotting/solar_cycle_example.py", line 35, in <module>
 
    noaa = ts.TimeSeries(f_noaa_indices, source='noaaindices').truncate(time_range)
 
AttributeError: 'list' object has no attribute 'truncate'
 
```
I replaced this offending line:
```python
noaa = ts.TimeSeries(f_noaa_indices, source='noaaindices').truncate(time_range)
```

With the following (based on the [GenericTimeSeries](https://docs.sunpy.org/en/stable/api/sunpy.timeseries.GenericTimeSeries.html) attributes summary):
```python
noaa = ts.TimeSeries(f_noaa_indices, source='noaaindices', time_range=time_range) 
```

But the `noaa.index` variable still contains the same time range as the downloaded file `f_noaa_indices`.

**Comment by hayesla** [comment]
> I don't understand this doc build error:
> 
> ```python
>   File "/root/project/examples/plotting/solar_cycle_example.py", line 35, in <module>
>  
>     noaa = ts.TimeSeries(f_noaa_indices, source='noaaindices').truncate(time_range)
>  
> AttributeError: 'list' object has no attribute 'truncate'
>  
> ```
> 
> I replaced this offending line:
> 
> ```python
> noaa = ts.TimeSeries(f_noaa_indices, source='noaaindices').truncate(time_range)
> ```
> 

this is strange - it works locally for me? 🤔

**Comment by mbobra** [comment]
> this is strange - it works locally for me? 🤔

Yeah it works locally for me too but I thought we should fix it because of the doc build error? I am not sure if we should address this error or not. If not, I'm happy to merge this.

**Comment by ayshih** [comment]
It should definitely not be merged if it's failing a build like this, because it'll just break every other CI build until we fix it.  From the error message, it looks as if `f_noaa_indices` has multiple results, which is why `ts.TimeSeries(f_noaa_indices, source='noaaindices') isn't a single `TimeSeries` object, but rather a `list` of them.  It may be a momentary freak-out from the Fido search, or it can point to some difference in the results that the CI receives.  I'm re-running the doc build now.

**Comment by nabobalis** [comment]
How do we deal with the history? Squash merge it?

**Comment by dstansby** [comment]
I think a rebase on to current master should fix the doc build.

**Comment by ayshih** [comment]
> I think a rebase on to current master should fix the doc build.

Yup, I fixed the doc build in #4203 

> How do we deal with the history? Squash merge it?

At minimum it should be squashed, but ideally this PR would be rebased into a couple of commits  (e.g., one for the example and a separate one for the removal of sample data).

**Comment by ehsteve** [comment]
I will do my best to rebase this.

**Comment by ehsteve** [comment]
Ugh, I give up. Just merge it and squash it. I hope that that does not delete @rutujasurve authorship though.

**Comment by dstansby** [comment]
I think I fixed the codestyle stuff...

File: sunpy/timeseries/sources/noaa.py [hunk 9]
```diff
@@ -202,7 +199,7 @@ def peek(self, **plot_args):
 
             import sunpy.timeseries
             import sunpy.data.sample
-            noaa = sunpy.timeseries.TimeSeries(sunpy.data.sample.NOAAPREDICT_TIMESERIES, source='NOAAPredictIndices')
+            noaa = sunpy.timeseries.TimeSeries('ftp://ftp.swpc.noaa.gov/pub/weekly/Predict.txt', source='NOAAPredictIndices')
```
Comment (dstansby) [inline sunpy/timeseries/sources/noaa.py:202]: I think this needs a remote data flag?

Comment (Cadair) [reply sunpy/timeseries/sources/noaa.py:202]: Looks like that might be causing the doc build failure.

Comment (ayshih) [reply sunpy/timeseries/sources/noaa.py:202]: ?
This is in a `plot` directive, not in a doctest.  There's no such thing as a remote-data flag.
The doc build is supposed to grab remote data.  Even if there were such a control flag, it'd be enabled for the doc build.

File: sunpy/timeseries/sources/noaa.py [hunk 10]
```diff
@@ -202,7 +199,7 @@ def peek(self, **plot_args):
 
             import sunpy.timeseries
             import sunpy.data.sample
```
Comment (ayshih) [inline sunpy/timeseries/sources/noaa.py:201]:
```suggestion
```

File: sunpy/timeseries/sources/noaa.py (hunk 9 above)
Comment (ayshih) [reply sunpy/timeseries/sources/noaa.py:202]: The doc build passed on a re-run, which suggests that switching to online retrieval may have made our online builds even more prone to failure.

**Comment by ayshih** [comment]
Having long URLs in the docstrings seems janky, since presumably that's not the recommended approach for users.  Some suggestions:

- Save canonical URLs in a variable somewhere so that it's easier for the user to input.
- Have the docstring examples use Fido instead of the URLs.
- Alternatively, if we don't want to entrain the docstring examples with Fido, add a Sphinx reference to the gallery example in the docstrings so that users are directed to the recommended approach.

**Comment by nabobalis** [comment]
I've sorted the history (I did screw it up tho).

**Comment by dstansby** [comment]
It looks like one of the changelog filenames needs updating to reflect this PR number.

**Comment by nabobalis** [comment]
That one points to the original PR that the first commit comes from, I kind of wanted to keep that way so it points to that PR in the changelog but I am happy to change it.

**Comment by ayshih** [comment]
I'm going to revise this PR

**Comment by ayshih** [comment]
Okay, updated.  I added multiple links to the gallery example, improved the changelog entries, fixed some other minor issues, and redistributed the changes to two commits.

As an aside, it looks like the "4135" changelog entry is being ignored by towncrier (because that PR is closed?).  We may need to change that to "4169" and then simply rely on this issue to provide the cross-reference.

**Comment by Cadair** [comment]
just ignore the failing check or stick no changelog on it, either way if you are doing it explicitly it's fine.

**Comment by Cadair** [comment]
> the file can be removed from the sample data server and a release of the sample data repo should be made in coordination with the next release which would include this PR.

Not yet it can't. We have to keep the file in the sample data repo to support all old versions of SunPy. There is no real disadvantage to doing so either, it wont be downloaded by sunpy versions post this PR.

Eventually it could be removed when we are happy to break sample data download for anyone who isn't running this version or above.

**Comment by ayshih** [comment]
I decided to rename the changelog entry so hopefully towncrier will accept it now.

**Comment by ehsteve** [comment]
> Having long URLs in the docstrings seems janky, since presumably that's not the recommended approach for users. Some suggestions:
> 
> * Save canonical URLs in a variable somewhere so that it's easier for the user to input.
> * Have the docstring examples use Fido instead of the URLs.
> * Alternatively, if we don't want to entrain the docstring examples with Fido, add a Sphinx reference to the gallery example in the docstrings so that users are directed to the recommended approach.

I agree with this comment. I wasn't sure what I wanted to do with the docstring. Having an example docstring that needs to download a file using Fido seemed even more janky especially because it is long code. If there was a one liner for Fido to return that URL that would be cool but seems like that requires a bit of work.

**Comment by ehsteve** [comment]
Thanks @ayshih for working on this! It has been a while since I've made a PR but I have made my fair share of contributions to SunPy BUT I was surprised at how hard it is now to get this through. I ran into a number of areas that I really had no idea how to address with the testing. I worry that we have raised a very high bar for anyone to contribute. One big issue was that it was difficult for me to setup the environments to duplicate the testing locally so I had to rely on the online tests which leads to muddying the commit history.

**Comment by Cadair** [comment]
> One big issue was that it was difficult for me to setup the environments to duplicate the testing locally

I would be interested in understanding this in more detail. Any issues that stop
```
conda/pip install tox
tox -e build_docs
```
working are a major issue. If you have the time to write up concrete issues you ran into in a separate issue that would be valuable.

**Comment by ehsteve** [comment]
> > One big issue was that it was difficult for me to setup the environments to duplicate the testing locally
> 
> I would be interested in understanding this in more detail. Any issues that stop
> 
> ```
> conda/pip install tox
> tox -e build_docs
> ```
> 
> working are a major issue. If you have the time to write up concrete issues you ran into in a separate issue that would be valuable.

That is one of the 12 different tests we run. I did not find using that tox build_docs command to be completely useful. I found I had to try to build the docs myself to better understand what I needed to do and actually see the changes I made to the docs. There are a number of new dependencies that docs need which were an annoying stumbling block (I recently had to wipe my setup). 

There were two tests I did not spend the time figuring out how to run locally; the pycodestyle and the changelog one.

Maybe we could provide instructions in the tests themselves that show contributors to run the tests locally.

**Comment by nabobalis** [comment]
> Maybe we could provide instructions in the tests themselves that show contributors to run the tests locally.

I am pretty sure we do.

**Comment by nabobalis** [comment]
@ehsteve I am very interested in your view on what you feel is over complicated and where we can improve our documentation on this matter.

**Comment by ehsteve** [comment]
Added a description of my issues in the [wiki](https://github.com/sunpy/sunpy/wiki/Testing-Feedback).

**Pull request merged**